lib/encoding.neon

```diff
@@ -0,0 +1,132 @@
+     -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1, -1,  62,  -1,  -1,  -1,  63,    % 0x20 - 0x2f
+     52,  53,  54,  55,  56,  57,  58,  59,  60,  61, -1,  -1,  -1,  64,  -1,  -1,    % 0x30 - 0x3f
+     -1,   0,   1,   2,   3,   4,   5,   6,   7,   8,  9,  10,  11,  12,  13,  14,    % 0x40 - 0x4f
+     15,  16,  17,  18,  19,  20,  21,  22,  23,  24, 25,  -1,  -1,  -1,  -1,  -1,    % 0x50 - 0x5f
+     -1,  26,  27,  28,  29,  30,  31,  32,  33,  34, 35,  36,  37,  38,  39,  40,    % 0x60 - 0x6f
+     41,  42,  43,  44,  45,  46,  47,  48,  49,  50, 51,  -1,  -1,  -1,  -1,  -1,    % 0x70 - 0x7f
+    % This table simply ignores these values.
+     -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1, -1,  -1,  -1,  -1,  -1,  -1,    % 0x80 - 0x8f
+     -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1, -1,  -1,  -1,  -1,  -1,  -1,    % 0x90 - 0x9f
+     -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1, -1,  -1,  -1,  -1,  -1,  -1,    % 0xa0 - 0xaf
+     -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1, -1,  -1,  -1,  -1,  -1,  -1,    % 0xb0 - 0xbf
+     -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1, -1,  -1,  -1,  -1,  -1,  -1,    % 0xc0 - 0xcf
+     -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1, -1,  -1,  -1,  -1,  -1,  -1,    % 0xd0 - 0xdf
+     -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1, -1,  -1,  -1,  -1,  -1,  -1,    % 0xe0 - 0xef
+     -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1, -1,  -1,  -1,  -1,  -1,  -1,    % 0xf0 - 0xff
+]
+
+%FUNCTION base64Encode(src: Bytes): String
+%|
+ |  Function: base64Encode
+ |
+ |  Encodes the provided Bytes into Base64 encoded data.
+ |
+ |  Parameters:
+ |      src - The source data to encode.
+ |
+ |%
+FUNCTION base64Encode(src: Bytes): String
+    VAR encoded: String := ""
+    VAR char1, char2: Number
+    VAR status: State := State.BYTE1
+
+    LET source: Array<Number> := src.toArray()
+
+    FOR i:= 0 TO src.size()-1 DO
+        LET char: Number := source[i]
+        CASE status
+            WHEN State.BYTE1 DO
+                encoded.append(Base64EncodeTable[bitwise.shiftRight32(char, 2)])
+                char1 := bitwise.and32(bitwise.shiftLeft32(char, 4), 0b00110000)
+                status := State.BYTE2
+            WHEN State.BYTE2 DO
+                encoded.append(Base64EncodeTable[bitwise.or32(char1, bitwise.shiftRight32(char, 4))])
+                char2 := bitwise.and32(bitwise.shiftLeft32(char, 2), 0b00111100)
+                status := State.BYTE3
+            WHEN State.BYTE3 DO
+                encoded.append(Base64EncodeTable[bitwise.or32(char2, bitwise.shiftRight32(char, 6))])
+                encoded.append(Base64EncodeTable[bitwise.and32(char, 0b00111111)])
+                status := State.BYTE1
+        END CASE
+    END FOR
+
+    CASE status
+        % In base64, when the data being encoded is NOT a multiple of 6 bits, needs to be padded.  The following pads the output data
+        % and signals we've hit the end of the encoded data when decoding...
```

Comment:
Why does this comment end with `...`? Did you have another thought to add here? When a reader encounters this, will they wonder what else you were going to say but didn't? Comments should be complete sentences, with a period at the end.